napari/utils/events/_tests/test_evented_list.py

```diff
@@ -0,0 +1,249 @@
+from napari.utils.events import EmitterGroup, EventedList, NestableEventedList
+from collections.abc import MutableSequence
+import pytest
+
+
+@pytest.fixture
+def regular_list():
+    return list(range(5))
+
+
+def flatten(_lst):
+    return (
+        flatten(_lst[0]) + (flatten(_lst[1:]) if len(_lst) > 1 else [])
+        if isinstance(_lst, MutableSequence)
+        else [_lst]
+    )
+
+
+@pytest.fixture(params=[EventedList, NestableEventedList])
+def test_list(request, regular_list):
+    test_list = request.param(regular_list)
+    test_list._events = []
+    test_list.events.connect(test_list._events.append)
+    return test_list
+
+
+@pytest.mark.parametrize(
```

Comment:
When I did the list model refactoring stuff in #1391 I added a test for event emission using `Mock` - see here https://github.com/sofroniewn/napari/blob/87910700153881903401c4f1d3f58285c94a674b/napari/utils/list/_tests/test_list_model.py

It seems like you've taken a different approach (which is totally fine), but can you explain if it tests the same stuff/ why you prefer this approach over `mock` - it would be a good way to standardize testing event emission etc in the repo too!!

Reply:
oh actually, I think I love what you did there!  but maybe we can do it slightly more generically such that we don't have to know names of the methods.  let me see if I can refactor tests here to use something like that

Reply:
Ok great, refactor and improve away!

Reply:
ok, used Mocks where possible.  I decided against creating a full fixture as you did there, instead using the [`wraps`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock) argument to Mock.  `test_list.events = Mock(wraps=test_list.events)`... That turns `test_list.events` into a pass-through object that you can spy on, and make assertions about what events were emitted, without explicitly saying what the event names should be.

(If you're wondering why I left that last test without a Mock, it's an idiosyncrasy/difference in the way events are emitted from nested children that makes it easier to test that way)

Reply:
(still one broken test for moving multiple items simultaneously)